rustup-init.sh

```diff
@@ -175,9 +172,10 @@ get_endianness() {
 }
 
 get_architecture() {
-    local _ostype _cputype
+    local _ostype _cputype _bitness _arch
     _ostype="$(uname -s)"
     _cputype="$(uname -m)"
+    _bitness="$(get_bitness)"
```

Comment:
This now runs on all platforms, which will fail on some with the following error.
```
head: /proc/self/exe: No such file or directory
rustup: unknown platform bitness
```
I suspect it was only run on linux for that reason.